### PR TITLE
Update to our Docker images

### PR DESCRIPTION
**Pull request by scholzj**
This PR is addressing 4 separate issues:
- #113 (Moves away from Fedora to CentOS - more enterprise OS)
- #147 (Uses Apache Kafka distribution for Zookeeper)
- #61 (Uses CMD in all images - makes it easy to override the commend when needed)
- #88 (Adds `ps` command to the images - as part of the CentOS7 move)

Additionally, it moves all the pure Docker images to single subdirectory `docker-images`.

### CODE REVIEW

**Comment by scholzj** [comment]
@ppatierno and @tombentley : Could one of you review it? Thanks ...

File: docker-images/kafka-connect/s2i/Dockerfile [hunk 1]
```diff
@@ -14,5 +14,4 @@ COPY ./scripts/ $S2I_HOME
 
 USER 1001:1001
 
-ENTRYPOINT [ ]
 CMD /opt/s2i/run
```
Comment (ppatierno) [inline docker-images/kafka-connect/s2i/Dockerfile]: Question (my ignorance) : Why not using the same syntax as in the other CMD ? I mean CMD ["/opt/s2i/run"]

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 RELEASE_VERSION ?= latest
 
-SUBDIRS=kafka-base zookeeper kafka kafka-connect kafka-connect/s2i cluster-controller topic-controller resources
+SUBDIRS=docker-images/kafka-base docker-images/zookeeper docker-images/kafka docker-images/kafka-connect docker-images/kafka-connect/s2i cluster-controller topic-controller resources
```
Comment (tombentley) [inline Makefile]: Is it possible to just specify `docker-images` here, and then override `SUBDIRS` in that directory to recurse though the various subdirs?

Comment (scholzj) [reply Makefile]: Please have a look at what I did there. It has some slight duplication. But ti allows to build only the docker-images without the controllers, which is IMHO an advantage.

File: docker-images/kafka-connect/s2i/Dockerfile (hunk 1 above)
Comment (scholzj) [reply docker-images/kafka-connect/s2i/Dockerfile]: This was leftover from before. Should be fixed now.

File: Makefile (hunk 2 above)
Comment (tombentley) [reply Makefile]: Yeah, that looks fine to me.

**Pull request merged**